File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.java
class Solution {
    public int[] productExceptSelf(int[] nums) {

        

        int[] list = new int[nums.length];

        for(int i=0; i<nums.length;i++){
            int j = 0;
            int pro = 1;

            while(j<nums.length){
                if(j!=i){
                    pro*=nums[j];

                }
                j++;
                
            }
            list[i]=pro;
            
        }

        return list;
        
    }
}  

File: Data Structures & Algorithms/maximum-product-subarray/submission-2.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:

        final = nums[0]
        
        mx = nums[0]
        mn = nums[0]

        for n in nums[1:]:

            old_mx = mx
            old_mn = mn 

            mx = max(n,old_mx*n,old_mn*n)
            mn = min(n,old_mx*n,old_mn * n)
            final = max(mx,final)

        return final
       
        


        
        